domain/schema/model/sql/0024-relation.sql

```diff
@@ -22,6 +22,24 @@ CREATE TABLE application_endpoint (
     REFERENCES charm_relation (uuid)
 );
 
+-- The application_extra_endpoint ties an application's
+-- definition of an extra binding to a space.
+CREATE TABLE application_extra_endpoint (
+    application_uuid TEXT NOT NULL,
+    space_uuid TEXT NOT NULL,
```

Comment:
I think I discussed with @SimonRichardson about adding the default binding to the `application` table, which would make this and its analogue in `application_endpoint` nullable.

Reply:
Correct, so this needs to be allowed to be nullable.

Reply:
Make the change, to remove `NOT NULL`, when you do put the default binding in application then. It's premature to do that now, as currently, all endpoints must have a space listed.

Reply:
adding the default endpoint binding